lib/dor/models/concerns/describable.rb

```diff
@@ -198,68 +200,74 @@ def add_constituent_relations(doc)
-        'sw_author_tesim'             => :main_author_w_date,
-        'sw_author_sort_ssi'          => :sw_sort_author,
-        'sw_language_ssim'            => :sw_language_facet,
-        'sw_language_tesim'           => :sw_language_facet,
-        'sw_genre_ssim'               => :sw_genre,
-        'sw_genre_tesim'              => :sw_genre,
-        'sw_format_ssim'              => :format_main,   # basically sw_typeOfResource_ssim
-        'sw_format_tesim'             => :format_main,   # basically sw_typeOfResource_tesim
-        'sw_topic_ssim'               => :topic_facet,
-        'sw_topic_tesim'              => :topic_facet,
-        'sw_subject_temporal_ssim'    => :era_facet,
-        'sw_subject_temporal_tesim'   => :era_facet,
-        'sw_subject_geographic_ssim'  => :geographic_facet,
-        'sw_subject_geographic_tesim' => :geographic_facet,
-        'mods_typeOfResource_ssim'    => [:term_values, :typeOfResource],
-        'mods_typeOfResource_tesim'   => [:term_values, :typeOfResource]
-      }
-      keys = mods_sources.keys.concat(%w( metadata_format_ssim ))
-      keys.each { |key|
-        solr_doc[key] ||= []     # initialize multivalue targts if necessary
+        sw_title_display: %w(sw_display_title_tesim),
+        main_author_w_date: %w(sw_author_ssim sw_author_tesim),
+        sw_sort_author: %w(sw_author_sort_ssi),
+        sw_language_facet: %w(sw_language_ssim sw_language_tesim),
+        sw_genre: %w(sw_genre_ssim sw_genre_tesim),
+        format_main: %w(sw_format_ssim sw_format_tesim),
+        topic_facet: %w(sw_topic_ssim sw_topic_tesim),
+        era_facet: %w(sw_subject_temporal_ssim sw_subject_temporal_tesim),
+        geographic_facet: %w(sw_subject_geographic_ssim sw_subject_geographic_tesim),
+        [:term_values, :typeOfResource] => %w(mods_typeOfResource_ssim mods_typeOfResource_tesim),
+        pub_year_sort_str: %w(sw_pub_date_sort_ssi),
+        pub_year_int: %w(sw_pub_date_sort_isi),
+        pub_year_display_str: %w(sw_pub_date_facet_ssi)
       }
 
-      solr_doc['metadata_format_ssim'] << 'mods'
-      begin
-        dc_doc = generate_dublin_core(include_collection_as_related_item: false)
-        # we excluding the generated collection relation here; we instead get the collection
-        # title from Dor::Identifiable.
-        dc_doc.xpath('/oai_dc:dc/*', oai_dc: XMLNS_OAI_DC).each do |node|
-          add_solr_value(solr_doc, "public_dc_#{node.name}", node.text, :string, [:stored_searchable])
-        end
-        creator = ''
-        dc_doc.xpath('//dc:creator', dc: XMLNS_DC).each do |node|
-          creator = node.text
-        end
-        title = ''
-        dc_doc.xpath('//dc:title', dc: XMLNS_DC).each do |node|
-          title = node.text
+      mods_sources.each_pair do |meth, solr_keys|
+        vals = meth.is_a?(Array) ? stanford_mods.send(meth.shift, *meth) : stanford_mods.send(meth)
+
+        next if vals.nil? || (vals.respond_to?(:empty?) && vals.empty?)
```

Comment:
I don't know what, if anything, the previous code was guarding against. I don't think it's blank values (or, if it is, the previous code wasn't guarding against that), but something bad coming back from stanford_mods?